addons/mail/static/src/models/message_list_view/message_list_view.js

```diff
@@ -6,11 +6,115 @@ import { attr, one } from '@mail/model/model_field';
+            );
+        },
+        /***
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsAtEnd() {
+            if (this.threadViewOwner.order === 'asc') {
+                return this.isAtBottom;
+            }
+            return this.isAtTop;
+        },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsAtStart() {
+            if (this.threadViewOwner.order === 'asc') {
+                return this.isAtTop;
+            }
+            return this.isAtBottom;
+        },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsAtTop() {
+            return this.scrollTop <= this.endThreshold;
+        },
+        /**
+         * @private
+         * @returns {Owl.ref}
+         */
+        _computeScrollableElement() {
+            if (
+                this.threadViewOwner.threadViewer &&
+                this.threadViewOwner.threadViewer.chatter &&
+                this.threadViewOwner.threadViewer.chatter.scrollPanelRef
+            ) {
+                return this.threadViewOwner.threadViewer.chatter.scrollPanelRef;
+            }
+            if (this.scrollRef) {
+                return this.scrollRef;
+            }
+        },
+    },
     fields: {
+        clientHeight: attr(),
         /**
          * States the OWL component of this message list view
          */
         component: attr(),
+        /**
+         * The margin that we use to detect that the scrollbar is a the end of
+         * the threadView.
+         */
+        endThreshold: attr({
+            default: 30,
+        }),
+        /**
+         * States whether the message list scroll position is at the bottom
+         * of the message list.
+         */
+        isAtBottom: attr({
+            compute: '_computeIsAtBottom',
+        }),
+        /**
+         * States whether the message list scroll position is at the end of
+         * the message list. Depending of the message list order, this could be
+         * the top or the bottom.
+         */
+        isAtEnd: attr({
+            compute: '_computeIsAtEnd',
+        }),
+        /**
+         * States whether the message list scroll position is at the start
+         * of the message list. Depending of the message list order, this could
+         * be the top or the bottom.
+         */
+        isAtStart: attr({
+            compute: '_computeIsAtStart',
+        }),
+        /**
+         * States whether the message list scroll position is at the top of
+         * the message list.
+         */
+        isAtTop: attr({
+            compute: '_computeIsAtTop',
+        }),
+        /**
+         * States the ref to the html node of the message list.
+         */
+        scrollRef: attr(),
+        /**
+         * States the scrollable element for the message list.
+         */
+        scrollableElement: attr({
```

Comment:
This should be renamed to `scrollableElementRef`, as it doesn't actually contain the element, it makes the code confusing to read.

Reply:
Sure !